### ISSUE TITLE
MCA milestone 1: Remote runc invocation using docker exec

### ISSUE THREAD

**Issue by corrieb** [vmware/vic#6859]
As simply as possible, let's demonstrate that the basic concept of starting a container VM with the runc binary allows us to remotely start, stop and interact with containers.

In terms of filesystem preparation, we can either use "docker build" to create a container image or use a pre-populated volume. Ultimately it doesn't matter, but the "docker build" option is closer to what we're ultimately looking for.

Step 1: Pick a container image: Can be as simple as hello world, although ideally it'll be something that runs for a reasonable period of time and produce some output so that we can attach to it.

Step 2: Using containerd, prepare the container filesystem inside of a docker image or volume in VIC. Take particular note of the configuration that's created and whether there's anything in there that will need to be modified for the target VM (IP address etc). Make sure that the runc binary is included.

Step 3: Start the image as a containerVM and use the Docker API to make exec calls to start and stop the container via runc. Once that's working, try making attach calls.

**Comment by vburenin** [comment]
The 3rd step happened to be a lot more complicated than just simple invocation of runc to run a container.
To be able to run containerd in detached mode, I needed to provide --console-socket which is a unix domain socket. There is another process needed that receives all data via this unix socket so process that is running by runc can dump its output out.

I ended up implementing socket management server that has a separate control channel that allows to create, list and serve those sockets on demand. All data that is received from these sockets is being dumped into a corresponding separate log file. The server is capable to provide an ability to attach to that socket console so another process can attach to that output and see what's going on.
